### PR TITLE
Add a PCD file creator.

### PR DESCRIPTION
**Pull request by at-wat**
This commit adds PCD file (PCD_v7) creator based on PointsProcessor class.

### CODE REVIEW

**Comment by SirVer** [comment]
Stuff changed a bit in the repo, so this needs updating. Sorry for that :/

Your implementation throws away a lot of points, was that intentional? I think this should look similar to  [the PLY writer](https://github.com/googlecartographer/cartographer/blob/master/cartographer/io/ply_writing_points_processor.cc) and retain all points in the output and also stream the points to disk.

File: cartographer/io/pcd_points_processor.cc [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  fs << "VERSION .7" << std::endl;
+  fs << "FIELDS x y z" << std::endl;
+  fs << "SIZE 4 4 4" << std::endl;
+  fs << "TYPE F F F" << std::endl;
+  fs << "COUNT 1 1 1" << std::endl;
+  fs << "WIDTH " << num << std::endl;
+  fs << "HEIGHT 1" << std::endl;
+  fs << "VIEWPOINT 0 0 0 1 0 0 0" << std::endl;
+  fs << "POINTS " << num << std::endl;
+  fs << "DATA ascii" << std::endl;
```
Comment (SirVer) [inline cartographer/io/pcd_points_processor.cc]: this should probably write out a binary file, otherwise size explodes really quickly.

**Comment by at-wat** [comment]
My first opinion was to use the PCD output directly as a map for localization.
However, considering other usages, full points in binary format would be better.

I'll take a look around the changes.

**Comment by at-wat** [comment]
I'll open a new PR.

**Pull request closed without merging**